src/Makefile

```diff
@@ -52,8 +52,11 @@ $(BUILD_DIR)/libutf8lex.so.$(UTF8LEX_LIBRARY_VERSION): $(OBJECT_FILES)
 	    -o $(BUILD_DIR)/libutf8lex.so.$(UTF8LEX_LIBRARY_VERSION)
 	chmod a+x $(BUILD_DIR)/libutf8lex.so.$(UTF8LEX_LIBRARY_VERSION)
 	cd $(BUILD_DIR) \
-	    && ln -s -f libutf8lex.so.$(UTF8LEX_LIBRARY_VERSION) \
-	        libutf8lex.so.$(UTF8LEX_MAJOR) \
+	    && if test "$(UTF8LEX_MAJOR)" != "$(UTF8LEX_LIBRARY_VERSION)" ; \
+                then \
+	            ln -s -f libutf8lex.so.$(UTF8LEX_LIBRARY_VERSION) \
+	                libutf8lex.so.$(UTF8LEX_MAJOR) ; \
+	        fi \
```

Comment:
In GitHub Actions, we do "make UTF8LEX_LIBRARY_VERSION=(branch-name)", where (branch-name) is not a version tag.  In such cases, the script to extract the major part of a semantic version gets the whole version string.  If we were to try to symlink, we'd fail on effectively `"ln -s libutf8.so.$(UTF8LEX_LIBRARY_VERSION) libutf8.so.$(UTF8LEX_LIBRARY_VERSION)"`.